NativeImageBuildStep fails if 'docker' is not installed

**Describe the bug**
On a Fedora 32 fresh installation where 'docker' command is not installed because 'podman' is only available to run containers, I found the following exception when compiled a simple HelloWorld project:

`$ mvn clean package -Pnative -Dquarkus.native.container-build=true
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< org.acme:code-with-quarkus >---------------------
[INFO] Building code-with-quarkus 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------

[INFO] 
[INFO] --- quarkus-maven-plugin:1.6.1.Final:build (default) @ code-with-quarkus ---
[INFO] [org.jboss.threads] JBoss Threads version 3.1.1.Final
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building native image source jar: /home/rmarting/Workspaces/ws-crs/quarkus-sample/target/code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar/code-with-quarkus-1.0.0-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from /home/rmarting/Workspaces/ws-crs/quarkus-sample/target/code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar/code-with-quarkus-1.0.0-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Checking image status quay.io/quarkus/ubi-quarkus-native-image:20.1.0-java11
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.124 s
[INFO] Finished at: 2020-07-27T12:57:40+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.6.1.Final:build (default) on project code-with-quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to pull builder image quay.io/quarkus/ubi-quarkus-native-image:20.1.0-java11
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:134)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:932)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] Caused by: java.io.IOException: Cannot run program "docker": error=2, No such file or directory
[ERROR] 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
[ERROR] 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
[ERROR] 	at io.quarkus.deployment.util.ProcessUtil.launchProcess(ProcessUtil.java:37)
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:131)
[ERROR] 	... 12 more
[ERROR] Caused by: java.io.IOException: error=2, No such file or directory
[ERROR] 	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
[ERROR] 	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
[ERROR] 	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
[ERROR] 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
[ERROR] 	... 15 more
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
`

I made an alias between docker and podman as:

`alias docker=podman`

However with the same failure.

**Expected behavior**
(Describe the expected behavior clearly and concisely.)

**Actual behavior**
(Describe the actual behavior clearly and concisely.)

**To Reproduce**
Steps to reproduce the behavior:
1. Create a simple helloworld sample
2. 'docker' cli not installed
3. Build project as: `mvn clean package -Pnative -Dquarkus.native.container-build=true`

**Environment :**
 - Output of `uname -a` or `ver`: 
`Linux redhat 5.7.9-200.fc32.x86_64 #1 SMP Fri Jul 17 16:23:37 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux` 
 - Output of `java -version`: 
`openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment 18.9 (build 11.0.8+10)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.8+10, mixed mode, sharing)`
 - GraalVM version (if different from Java): 
`GraalVM Updater 20.1.0` 
 - Quarkus version or git rev: 
`1.6.1.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
`Maven home: /usr/share/maven
Java version: 11.0.8, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-11-openjdk-11.0.8.10-2.fc32.x86_64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.7.9-200.fc32.x86_64", arch: "amd64", family: "unix"
`

--- a/docs/src/main/asciidoc/building-native-image.adoc
+++ b/docs/src/main/asciidoc/building-native-image.adoc
@@ -298,19 +298,18 @@ failsafe.
 If the process cannot find the native image for some reason, or you want to test a native image that is no longer in the
 target directory you can specify the executable with the `-Dnative.image.path=` system property.
 
-== Creating a container
+[#container-runtime]
+== Creating a Linux executable without GraalVM installed
 
-IMPORTANT: Before going further, be sure to have a working container runtime (Docker, podman) environment. If you use Docker 
+IMPORTANT: Before going further, be sure to have a working container runtime (Docker, podman) environment. If you use Docker
 on Windows you should share your project's drive at Docker Desktop file share settings and restart Docker Desktop.
 
-You can run the application in a container using the JAR produced by the Quarkus Maven Plugin.
-However, in this guide we focus on creating a container image using the produced native executable.
+Quite often one only needs to create a native Linux executable for their Quarkus application (for example in order to run in a containerized environment) and would like to avoid
+the trouble of installing the proper GraalVM version in order to accomplish this task (for example, in CI environments it's common practice
+to install as little software as possible).
 
-image:containerization-process.png[Containerization Process]
-
-By default, the native executable is tailored for your operating system (Linux, macOS, Windows etc).
-Because the container may not use the same _executable_ format as the one produced by your operating system,
-we will instruct the Maven build to produce an executable from inside a container:
+To this end, Quarkus provides a very convenient way of creating a native Linux executable by leveraging a container runtime such as Docker or podman.
+The easiest way of accomplishing this task is to execute:
 
 [source, shell]
 ----
@@ -328,10 +327,40 @@ You can also select the container runtime to use with:
 ./mvnw package -Pnative -Dquarkus.native.container-runtime=podman
 ----
 
+
 These are normal Quarkus config properties, so if you always want to build in a container
-it is recommended you add these to your `application.properties` so you do not need to specify them every time.
+it is recommended you add these to your `application.properties` in order to avoid specifying them every time.
 
 ====
+
+== Creating a container
+
+=== Using the container-image extensions
+
+By far the easiest way to create a container-image from your Quarkus application is to leverage one of the container-image extensions.
+
+If one of those extensions is present, then creating a container image for the native executable is essentially a matter of executing a single command:
+
+[source, shell]
+----
+./mvnw package -Pnative -Dquarkus.native.container-build=true -Dquarkus.container-image.build=true
+----
+
+* `quarkus.native.container-build=true` allows for creating a Linux executable without GraalVM being installed (and is only necessary if you don't have GraalVM installed locally or your local operating system is not Linux)
+* `quarkus.container-image.build=true` instructs Quarkus to create a container-image using the final application artifact (which is the native executable in this case)
+
+See the link:container-image[Container Image guide] for more details.
+
+=== Manually
+
+You can run the application in a container using the JAR produced by the Quarkus Maven Plugin.
+However, in this section we focus on creating a container image using the produced native executable.
+
+image:containerization-process.png[Containerization Process]
+
+When using a local GraalVM installation, the native executable targets your local operating system (Linux, macOS, Windows etc).
+However, as a container may not use the same _executable_ format as the one produced by your operating system,
+we will instruct the Maven build to produce an executable by leveraging a container runtime (as described in <<#container-runtime,this section>>):
 
 The produced executable will be a 64 bit Linux executable, so depending on your operating system it may no longer be runnable.
 However, it's not an issue as we are going to copy it to a container.
@@ -375,9 +404,9 @@ And finally, run it with:
 docker run -i --rm -p 8080:8080 quarkus-quickstart/getting-started
 ----
 
-NOTE: Interested by tiny Docker images, check the {quarkus-images-url}/master/distroless[distroless] version.
+NOTE: If you are interested in tiny Docker images, check the {quarkus-images-url}/master/distroless[distroless] version.
 
-== Creating a container with a multi-stage Docker build
+=== Using a multi-stage Docker build
 
 The previous section showed you how to build a native executable using Maven, but implicitly required that the proper GraalVM version be installed on the building machine (be it your local machine or your CI/CD infrastructure).
 
@@ -442,9 +471,9 @@ Please see link:native-and-ssl#working-with-containers[our Using SSL With Native
 ====
 
 [[configuration-reference]]
-== Configuring the Native Image
+== Configuring the Native Executable
 
-There are a lot of different configuration options that can affect how the native image is generated.
+There are a lot of different configuration options that can affect how the native executable is generated.
 These are provided in `application.properties` the same as any other config property.
 
 The properties are shown below:

--- a/docs/src/main/asciidoc/native-and-ssl.adoc
+++ b/docs/src/main/asciidoc/native-and-ssl.adoc
@@ -190,6 +190,7 @@ will ensure that the certificates of `/tmp/mycerts` are baked into the native bi
 The file containing the custom TrustStore does *not* have to be present at runtime as its content has been baked into the native binary.
 ====
 
+[#working-with-containers]
 === Working with containers
 
 No special action needs to be taken when running the native binary in a container. If the native binary was properly built with the custom TrustStore

